token/program-2022/src/extension/mod.rs

```diff
@@ -693,79 +695,87 @@ impl From<ExtensionType> for [u8; 2] {
+            ExtensionType::ImmutableOwner => Ok(pod_get_packed_len::<ImmutableOwner>()),
             ExtensionType::ConfidentialTransferMint => {
-                pod_get_packed_len::<ConfidentialTransferMint>()
+                Ok(pod_get_packed_len::<ConfidentialTransferMint>())
             }
             ExtensionType::ConfidentialTransferAccount => {
-                pod_get_packed_len::<ConfidentialTransferAccount>()
+                Ok(pod_get_packed_len::<ConfidentialTransferAccount>())
             }
-            ExtensionType::DefaultAccountState => pod_get_packed_len::<DefaultAccountState>(),
-            ExtensionType::MemoTransfer => pod_get_packed_len::<MemoTransfer>(),
-            ExtensionType::NonTransferable => pod_get_packed_len::<NonTransferable>(),
-            ExtensionType::InterestBearingConfig => pod_get_packed_len::<InterestBearingConfig>(),
-            ExtensionType::CpiGuard => pod_get_packed_len::<CpiGuard>(),
-            ExtensionType::PermanentDelegate => pod_get_packed_len::<PermanentDelegate>(),
-            ExtensionType::NonTransferableAccount => pod_get_packed_len::<NonTransferableAccount>(),
-            ExtensionType::TransferHook => pod_get_packed_len::<TransferHook>(),
-            ExtensionType::TransferHookAccount => pod_get_packed_len::<TransferHookAccount>(),
+            ExtensionType::DefaultAccountState => Ok(pod_get_packed_len::<DefaultAccountState>()),
+            ExtensionType::MemoTransfer => Ok(pod_get_packed_len::<MemoTransfer>()),
+            ExtensionType::NonTransferable => Ok(pod_get_packed_len::<NonTransferable>()),
+            ExtensionType::InterestBearingConfig => {
+                Ok(pod_get_packed_len::<InterestBearingConfig>())
+            }
+            ExtensionType::CpiGuard => Ok(pod_get_packed_len::<CpiGuard>()),
+            ExtensionType::PermanentDelegate => Ok(pod_get_packed_len::<PermanentDelegate>()),
+            ExtensionType::NonTransferableAccount => {
+                Ok(pod_get_packed_len::<NonTransferableAccount>())
+            }
+            ExtensionType::TransferHook => Ok(pod_get_packed_len::<TransferHook>()),
+            ExtensionType::TransferHookAccount => Ok(pod_get_packed_len::<TransferHookAccount>()),
             ExtensionType::ConfidentialTransferFeeConfig => {
-                pod_get_packed_len::<ConfidentialTransferFeeConfig>()
+                Ok(pod_get_packed_len::<ConfidentialTransferFeeConfig>())
             }
             ExtensionType::ConfidentialTransferFeeAmount => {
-                pod_get_packed_len::<ConfidentialTransferFeeAmount>()
+                Ok(pod_get_packed_len::<ConfidentialTransferFeeAmount>())
             }
-            ExtensionType::MetadataPointer => pod_get_packed_len::<MetadataPointer>(),
+            ExtensionType::MetadataPointer => Ok(pod_get_packed_len::<MetadataPointer>()),
+            ExtensionType::Metadata => Err(ProgramError::InvalidArgument),
```

Comment:
What do you think about adding a method along the lines of `get_account_type` that is something like:
```
pub fn sized(&self) -> bool {
    match self {
        ExtensionType::Metadata => false,
        _ => true, // or list extensions explicitly to force people adding new extensions to handle
    }
}
```

And then the method here becomes:
```
pub fn get_type_len(&self) -> Result<usize, ProgramError> {
    if !self.sized() {
        return Err(ProgramError::InvalidArgument);
    }
    Ok(match self {
        ...
        ExtensionType::Metadata => unreachable!(),
    })
}
```
I feel like this would make it slightly more obvious what's going on with extensions like Metadata. Flipside, it is kind of bloaty. Thoughts?

Reply:
I'm down for it, feels a lot like "code as documentation", so even though it isn't necessary, it clarifies what's going on